src/registrar/views/application.py

```diff
@@ -0,0 +1,421 @@
+
+        if self.has_pk():
+            id = self.storage["application_id"]
+            try:
+                self._application = DomainApplication.objects.get(
+                    creator=self.request.user,  # type: ignore
+                    pk=id,
+                )
+                return self._application
+            except DomainApplication.DoesNotExist:
+                logger.debug("Application id %s did not have a DomainApplication" % id)
+
+        self._application = DomainApplication.objects.create(
+            creator=self.request.user,  # type: ignore
+        )
+        self.storage["application_id"] = self._application.id
+        return self._application
+
+    @property
+    def storage(self):
+        # marking session as modified on every access
+        # so that updates to nested keys are always saved
+        self.request.session.modified = True
+        if self.prefix not in self.request.session:
+            self.request.session[self.prefix] = {}
+        return self.request.session[self.prefix]
+
+    @storage.setter
+    def storage(self, value):
+        self.request.session[self.prefix] = value
+        self.request.session.modified = True
+
+    @storage.deleter
+    def storage(self):
+        del self.request.session[self.prefix]
+        self.request.session.modified = True
+
+    def done(self):
+        """Called when the user clicks the submit button, if all forms are valid."""
+        self.application.submit()  # change the status to submitted
+        self.application.save()
+        logger.debug("Application object saved: %s", self.application.id)
+        return redirect(reverse(f"{self.URL_NAMESPACE}:finished"))
+
+    def from_model(self, attribute: str, default, *args, **kwargs):
+        """
+        Get a attribute from the database model, if it exists.
+
+        If it is a callable, call it with any given `args` and `kwargs`.
+
+        This method exists so that we can avoid needlessly creating a record
+        in the database before the wizard has been saved.
+        """
+        if self.has_pk():
+            if hasattr(self.application, attribute):
+                attr = getattr(self.application, attribute)
+                if callable(attr):
+                    return attr(*args, **kwargs)
+                else:
+                    return attr
+            else:
+                raise AttributeError("Model has no attribute %s" % str(attribute))
+        else:
+            return default
+
+    def get(self, request, *args, **kwargs):
+        """This method handles GET requests."""
+
+        current_url = resolve(request.path_info).url_name
+
+        # if user visited via an "edit" url, associate the id of the
+        # application they are trying to edit to this wizard instance
+        if current_url == self.EDIT_URL_NAME and "id" in kwargs:
+            self.storage["application_id"] = kwargs["id"]
+
+        # if accessing this class directly, redirect to the first step
```

Comment:
**suggestion (non-blocking)**: I think that this comment could be even more explicit about how this won't apply to the rest of our views which are sub-classes (?) and so GET requests for the other views won't return us to the first step?

Reply:
Inserted longer comment here.